root/meta/autoloading/ROOT-12378/exec_testtypedef.C

```diff
@@ -0,0 +1,23 @@
+{
+  std::string name("TestMissingETBase::Types::jetlink_t");
+  auto cl = TClass::GetClass(name.c_str());
+  if (!cl) {
+     cerr << "Error the first call to TClass::GetClass for '" << name << "' returned : "
+          << cl << '\n';
+
+     auto cci = gInterpreter->CheckClassInfo(name.c_str(), kTRUE, kTRUE);
+     std::cerr << "CheckClassInfo returned    : " << cci << '\n';
+
+     std::string normalizedName;
+     {
+            TInterpreter::SuspendAutoLoadingRAII autoloadOff(gInterpreter);
+            TClassEdit::GetNormalizedName(normalizedName, name);
+     }
+     cerr << "Normalized named is  : " << normalizedName << '\n';
+     cerr << "Second attempt: " << TClass::GetClass(name.c_str())
+          << '\n';
+     return 1;
+  } else {
+     return 0;
```

Comment:
Maybe reverse the `if` block to make the test more readable, as in:
```c++
if (cl)
   return 0;
// [rest of the code here]
```

Reply:
Note early return is 'broken' for unnamed script so apply 'strictly' the proposed change fails: it still executes the lines following the if+return statement (and sometimes skip the next statement).